openapi-model/src/main/scala/sttp/apispec/openapi/validation/OpenAPICompatibilityIssue.scala

```diff
@@ -0,0 +1,175 @@
+package sttp.apispec.openapi.validation
+
+import sttp.apispec.SecurityRequirement
+import sttp.apispec.openapi.{ParameterStyle, ResponsesKey}
+import sttp.apispec.validation.SchemaCompatibilityIssue
+
+sealed abstract class OpenAPICompatibilityIssue {
+  def description: String
+
+  override def toString: String = description
+  protected def issuesRepr(issues: List[OpenAPICompatibilityIssue]): String =
+    issues.iterator
+      .map(i => s"- ${i.description.replace("\n", "\n  ")}") // indent
+      .mkString("\n")
```

Comment:
last thing here: can we add some tests that would verify the rendered representation of the issues? I know it might be a fragile test, but it's an important part of the "DX". So maybe we could have one or two cases where in addition to checking the issue representation, we'd also check the rendered form.